lib/src/git.rs

```diff
@@ -790,37 +810,44 @@ fn delete_git_ref(
 }
 
 fn update_git_ref(
-    git_repo: &git2::Repository,
+    git_repo: &gix::Repository,
     git_ref_name: &str,
-    old_oid: Option<Oid>,
-    new_oid: Oid,
+    old_oid: Option<gix::ObjectId>,
+    new_oid: gix::ObjectId,
 ) -> Result<(), FailedRefExportReason> {
     match old_oid {
         None => {
             if let Ok(git_repo_ref) = git_repo.find_reference(git_ref_name) {
                 // The branch was added in jj and in git. We're good if and only if git
                 // pointed it to our desired target.
-                if git_repo_ref.target() != Some(new_oid) {
+                if git_repo_ref.inner.target.try_id() != Some(&new_oid) {
                     return Err(FailedRefExportReason::AddedInJjAddedInGit);
                 }
             } else {
                 // The branch was added in jj but still doesn't exist in git, so add it
                 git_repo
-                    .reference(git_ref_name, new_oid, false, "export from jj")
-                    .map_err(FailedRefExportReason::FailedToSet)?;
+                    .reference(
+                        git_ref_name,
+                        new_oid,
+                        gix::refs::transaction::PreviousValue::MustNotExist,
```

Comment:
Nice that we're getting a stricter behavior compared to libgit2.

Reply:
Actually `false` means `MustNotExist` in libgit2, but yeah, gix provides stricter and readable API.